### ISSUE TITLE
Gasland

### ISSUE THREAD

**Issue by JeffreyStrahm** [siggame/MegaMinerAI-Game-Ideas#55]
(inspired by the game system gasland)
# **The game:**
## **TLDR**
- Outrun your opponent to win.
- 15x50 with road and different terrain types.
- customizable units!
- 3 types of units, bike, car, truck.
- Gain crowd favor to purchase new units.
- Stay in the lead to get competitive points to win.

## **story**
You are the manager of one of the two greatest racing teams to ever exist on earth. Today, you and your fierce rival will be battling it out in the race to end all races. The winner gets to retire in luxury on mars. The catch is that this race will be heavily controlled by crowd favor as the network wants to make this the greatest race of all time. Be careful as people tend to favor the underdog.

## **Victory:**
The game continues until either player has no vehicles active on the map, or until x turns have passed.
If x turns have passed, the player with the most competitive points wins.
Otherwise, it is whoever has the most value of vehicles left.
Otherwise, it is whoever has the most crowd favor.
Otherwise, it is a coin flip.

## **Map:**
The map is a 15 by 50 road. Since you only have to turn left, you don't need to too much about maneuvering. The final 5 rightmost tiles are unreachable. Every turn every unit and hazard moves to the left 2 spaces, if they move off the map they are removed from the game. Hazards affect units if they are ever atop them. Slowing hazards reduce units movement next turn while damaging hazards will apply their effects as they move past.

The top and bottom 2-4 tiles of the map are always rough terrain that takes 2 moves to move through. As hazards shift to the left off the screen, new hazards will spawn in the rightmost tiles. This will be covered more in **Hazards** and their avoidance covered in **movement**.

## **crowd favor and competition points:**
If there is one thing to know about gaslands, it is that everyone loves the underdog. This is why crowd favor (CF) is gain more rapidly when a player is losing. Whenever a player's unit dies, they gain half of it's cost back. Players also gain CF the closer their units are to the back of the map, and average. The losing player will gain at most 1.5 times the normal income of CF. The other player will always gain 1 times that amount.

competition points (CP) are gain the closer their units are to the front of the map, specifically the lead most unit. They are also only gained by the player in the lead in this fashion. CP are also gain each time an opposing unit is killed. The amount gained is one-tenth it's crowd favor value.

Note: it is possible for a player to be in the lead and gaining CP and gain a bonus number of CF points.

## **Hazards:**
In the game, there are two types of hazards, movement hazards, and damaging hazards.

**movement hazards**: movement hazards cost extra moves to move through. For now, this will just be referred to as rough terrain. Rough terrain costs 2 moves to move through and will drag a unit back an additional space when the map is shifting. They can be created by naturally spawning in the right portion of the map usually on the sides and uncommonly in the center. They may also be created in the wake of damaging hazards and weapons.

**damaging hazards**: damaging hazards are created by weapons, and may rarely spawn on the right side of the map, but only up to once per shift. Damaging tiles range in damage amount but do not affect movement unless a movement hazard exists on the same tile.

**mixed hazards**: Did I say two types of hazards? I didn't technically lie. This is the case where they both co-exist on the same tile. There is nothing else that is special.

## **Movement:**
Each turn every unit gets x moves. Moves may be used to move onto an adjacent tile. Moves onto rough terrain cost 2 movements. If a player moves onto another unit, that unit is pushed in the direction the unit came from and the pusher takes its place. This deals X damage to both cars. Terrain effects are applied normally.

## **Units:**
There are 3 types of units in the game. Bikes are the fastest units with little heath and few gear points (GP). They are also the cheapest. Cars are the middle unit with a balanced speed, GP, health, and cost. Trucks are the most expensive units, and are also the slowest, but have the highest health and GP pool.

| Type | HP | speed | GP | cost |
|----|----|----|----|----|
|**Bike**|5|6|3|4CF|
|**Car**|10|5|6|6CF|
|**Truck**|20|4|12|10CF|

## *Gear + combat:**
There are three types of gear: weapons, utility, and passive. each gear uses a certain amount of GP to put on a vehicle and costs a small amount of CF. Each unit is allowed one weapon gear, one utility gear, and as many passive gear as they can fit. Utility gear can only be used every other turn. utility gear doesn't cost a action to use.

Please note all of these are ideas, not guarantees. Many will probably be cut, especially from utility....

weapon gear:

| Name | GP | Cost | Damage | Range |
|----|----|----|----|----|
|**Small machine gun**|1|0CF|1|2|
|**Turret**|2|1CF|2|3|
|**Cannon**|4|3CF|5|6|
|**Gauss rifle**|8|5CF|9|12|

Utility gear:

| Name | GP | Cost | Range | Effect |
|----|----|----|----|----|
|**Net gun**|2|0CF|3|_Reduces a vehicle's movement for next turn by 2 (minimum negative movement)_|
|**Grappling hook**|2|1CF|8|_moves the target and user together one tile, collisions apply normally with target movement favored_|
|**Molotov Cocktail**|1|1CF|10|_creates a light damaging hazard tile_|
|**Caltrops**|2|2CF|10|_creates a damaging hazard tile_|
|**Mine launcher**|4|4CF|10|_creates a serious damaging hazard tile_|
|**Oil slick**|1|1CF|10|_creates a light movement hazard tile_|
|**Deployable speedbump**|2|2CF|10|_creates a movement hazard tile_|
|**Glue launcher**|4|4CF|10|_creates a serious movement hazard tile_|
|**Faulty hover rig**|1|0CF|0|_allows units to ignore hazards for one turn_|
|**Translocation device**|1|3CF|0|_swaps places with the next unit to use this device_|
|**Nitro boost light**|1|0CF|0|_adds 1 movement to the user_|
|**Nitro boost XL**|2|1CF|0|_adds 3 movements to the user_|

Note: the if you are the second person to use the translocation device, the people who have used it "reset" and the next person to use it must wait for someone else to use it to swap. This does allow you to swap with opponents.

Passive gear:

| Name | GP | CF | Effect |
|----|----|----|----|
|**Cheap armor plating**|1|0CF|_adds 2 health to the vehicle_|
|**Armor plating**|2|1CF|_adds 5 health to the vehicle_|
|**Heavy armor plating**|4|2CF|_adds 10 health to the vehicle_|
|**Working engine**|2|2CF|_adds 2 movement to the vehicle_|
|**Expensive engine**|1|5CF|_adds 4 movement to the vehicle_|
|**Working hover rig**|1|4CF|_You no longer suffer from terrain effects_|
|**Flix capacitor**|2|4CF|_Spawn time is next turn, no matter the location chosen_|
|**Rookie weapon overclock**|1|1CF|_increases damage done by weapons by 1_|
|**weapon overclock**|2|1CF|_increases damage done by weapons by 2_|
|**masterful weapon overclock**|4|2CF|_increases weapon damage done by 5_|
|**spring loaded bumpers**|4|2CF|_half collision damage_|

Note: you may only use one of each gear type. That means only one gear that adds to health, one that adds to movement, one that adds to damage etc.

## **Spawning:**
Whenever a unit is purchased, an empty tile must be selected. The further up the track that tile is, the longer it takes to spawn. A unit spawning will not stop a player from losing the game. It doesn't cost anything extra to spawn it further up. Should the tile be filled, the nearest empty tile favoring moving to the left, then north, then south, and then finally right will be it's spawn location. Spawn time for the unit will be a minimum of one turn.

**Comment by JeffreyStrahm** [comment]
There is definitely too many different things with this. Everything will need to be trimmed down.

**Comment by TwinNumberOne** [comment]
so customizable death race? Where the AI needs to navigate the obstacles and take out their opponent, where it is almost always guaranteed to be closed due to the CF points.

**Comment by JeffreyStrahm** [comment]
Can someone add the new mega miner tag to this idea?
